mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/HandlerMappingRegistry.java

```diff
@@ -0,0 +1,27 @@
+package com.interface21.webmvc.servlet.mvc.tobe;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import jakarta.servlet.http.HttpServletRequest;
+
+import com.interface21.context.container.Container;
+
+public class HandlerMappingRegistry {
+
+    private final List<HandlerMapping> handlerMappings;
+
+    public HandlerMappingRegistry() {
+        Container container = Container.getInstance();
+        handlerMappings = container.getInstancesOf(HandlerMapping.class);
+        handlerMappings.forEach(HandlerMapping::initialize);
+    }
+
+    public Object getHandler(final HttpServletRequest request) {
+        return handlerMappings.stream()
+                .flatMap(handlerMapping -> Stream.ofNullable(handlerMapping.getHandler(request)))
```

Comment:
로직이 잘 이해가 안되서 그런데 flatMap을 사용해야하는 이유가 있나요?
map으로는 안되나요?

Reply:
`getHandler` 메서드에서 null을 반환할 수 있기 때문에 optional과 flatMap을 사용해서 null을 없애고 있었습니다. 

해당 부분은 아래의 코드와 동일해요. 

```suggestion
                .map(handlerMapping -> handlerMapping.getHandler(request))
                .filter(Objects::nonNull)
```

제가 생각해도 flatMap을 사용하면 이해하기 어려운 것 같아요 ㅎㅎ... 

다음 단계에서는 map과 filter를 사용한 코드로 변경해볼게요~!